examples/server/server.cpp

```diff
@@ -541,12 +541,16 @@ struct completion_token_output {
     json to_json(bool post_sampling_probs) const {
         json probs_for_token = json::array();
         for (const auto & p : probs) {
-            std::string txt(p.txt);
+            // If the predicted token id is the same as this.tok, then we use the text_to_send instead
+            // of the detokenized token. This is to avoid a mismatch between the text tokens where
+            // the text_to_send token may include a leading whitespace character but the detokenized
+            // token would not.
+            std::string txt = tok == p.tok ? text_to_send : p.txt;
```

Comment:
Hi, thanks for working on this. Just wanted to mention that fixing the whitespace only for the first token in `top_logprobs` is not going to fix the use case I'm working with - the tokens in the other `top_logprobs` entries need to have the correct whitespace as they did before #10783. Sorry if the issue description was unclear.

Reply:
Ah, thanks for the clarification. I'll revert and look into this again now.